node/silkworm/common/log.cpp

```diff
@@ -38,7 +38,14 @@ std::mutex log_::log_mtx_;
 
 std::ostream& log_::header_(LogLevel level) {
     log_streams_ << kLogTags_[static_cast<int>(level)] << "["
+#if defined(_WIN32)
```

Comment:
This is not portable. We don't know if abseil is huntered or installed via vcpkg

Reply:
This is indeed portable as will work no matter how Abseil is installed, since the change is to avoid passing the first parameter, which can be aliased tp std::string_view or absl::string_view.
However it is true that the change may not be necessary if abseil is *not* installed via vcpkg.
I can try removing Abseil from vcpkg, and build again.

Reply:
I believe it may be hunter causing abseil to be installed via vcpkg.

Reply:
I don't think so. hunter and vcpkg are completely separate and unaware of each other.
What is possibile instead is **cmake's find_package** picks the package installed in vcpkg instead of the one downloaded and built by hunter. Most likely you have enabled vcpkg integration with visual studio

Reply:
OK, I think I'm going to give up on doing the windows build. Maybe I'll try WSL.

Reply:
Before giving up ... can you try disable vcpkg integration ?
`vcpkg integate remove`

Reply:
I tried generating the solution files using cmake-gui and then build that in Visual Studio, and get this error:

```
22>LINK : fatal error LNK1181: cannot open input file 'GMP_LIBRARY-NOTFOUND.obj' 
```

which is weird because I have this in the cmake config:

![image](https://user-images.githubusercontent.com/1496907/129450758-0779b495-c3d9-40bd-a53e-1d1e72624ef3.png)

Reply:
If you look at our [appveyor configuration](https://github.com/torquem-ch/silkworm/blob/master/appveyor.yml) you'll see you need to set PATH to include vcpkg directories

Reply:
thanks, I'll try removing vcpkg integration.
Unfortunately I also upgraded to the latest VS2019 version and now I have:

```
1> [CMake] C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Source\lib\keccak\keccakf800.c : error C2220: the following warning is treated as an error [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release-prefix\src\ethash-Release-build\lib\keccak\keccak.vcxproj] [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release.vcxproj]
1> [CMake] C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Source\lib\keccak\keccakf800.c : warning C5072: ASAN enabled without debug information emission. Enable debug info for better ASAN error reporting [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release-prefix\src\ethash-Release-build\lib\keccak\keccak.vcxproj] [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release.vcxproj]
1> [CMake] C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Source\lib\keccak\keccak.c : error C2220: the following warning is treated as an error [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release-prefix\src\ethash-Release-build\lib\keccak\keccak.vcxproj] [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release.vcxproj]
1> [CMake] C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Source\lib\keccak\keccak.c : warning C5072: ASAN enabled without debug information emission. Enable debug info for better ASAN error reporting [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release-prefix\src\ethash-Release-build\lib\keccak\keccak.vcxproj] [C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1\Build\ethash\Build\ethash-Release.vcxproj]
1> [CMake] C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'C:\greg\.hunter\_Base\6c9b2bc\4a379ff\9d920f1
```

Reply:
Edit CMakeSettings.json and change `addressSanitizerEnabled` to `false`
(Is an issue we're already aware of)

Reply:
Thanks, let me try these two things!

Reply:
Hi @AndreaLanfranchi, thanks a lot, with those changes I was able to build silkworm on windows, and indeed the change from this pull request is not needed. I will close it.